doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,248 @@
+# Overlay ECMP with BFD monitoring
+## High Level Design Document
+### Rev 1.0
+
+# Table of Contents
+
+  * [Revision](#revision)
+
+  * [About this Manual](#about-this-manual)
+
+  * [Scope](#scope)
+
+  * [Definitions/Abbreviation](#definitionsabbreviation)
+ 
+  * [1 Requirements Overview](#1-requirements-overview)
+    * [1.1 Usecase](#11-usecase)
+    * [1.2 Functional requirements](#12-functional-requirements)
+    * [1.3 CLI requirements](#13-cli-requirements)
+    * [1.4 Warm Restart requirements ](#14-warm-restart-requirements)
+  * [2 Modules Design](#2-modules-design)
+    * [2.1 Config DB](#21-config-db)
+    * [2.2 App DB](#22-app-db)
+    * [2.3 Module Interaction](#23-module-interaction)
+    * [2.4 Orchestration Agent](#24-orchestration-agent)
+    * [2.5 Monitoring and Health](#25-monitoring-and-health)
+    * [2.6 BGP](#26-bgp)
+    * [2.7 CLI](#27-cli)
+
+###### Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 09/09/2021  |     Prince Sunny   | Initial version                   |
+| 1.0 | 09/13/2021  |     Prince Sunny   | Revised                           |
+
+# About this Manual
+This document provides general information about the Vxlan Overlay ECMP feature implementation in SONiC with BFD support. This is an extension to the existing VNET Vxlan support as defined in the [Vxlan HLD](https://github.com/Azure/SONiC/blob/master/doc/vxlan/Vxlan_hld.md)
+# Scope
+This document describes the high level design of the Overlay ECMP feature and associated BFD support. General BFD support and configurations are beyond the scope of this document. 
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+|                          |                                |
+|--------------------------|--------------------------------|
+| BFD                      | Bidirectional Forwarding Detection       |
+| VNI                      | Vxlan Network Identifier       |
+| VTEP                     | Vxlan Tunnel End Point         |
+| VNet                     | Virtual Network                |
+
+# 1 Requirements Overview
+
+## 1.1 Usecase
+
+![](https://github.com/Azure/SONiC/blob/master/images/vxlan_hld/OverlayEcmp_UseCase.png)
+
+## 1.2 Functional requirements
+
+At a high level the following should be supported:
+
+- Configure ECMP with Tunnel Nexthops (IPv4 and IPv6)
+- Tunnel Endpoint monitoring via BFD
+- Add/Withdraw Nexthop based on Tunnel or Endpoint health
+
+## 1.3 CLI requirements
+- User should be able to show the BFD session
```

Comment:
How does an user configure BFD sessions? CLI requirement mentions only about show CLI for bfd sessions.

Reply:
Updated in PR - https://github.com/Azure/SONiC/pull/880